spec/index.bs

```diff
@@ -6516,108 +6552,35 @@ GPUQueue includes GPUObjectBase;
                     <div class=queue-timeline>
                         1. For each |commandBuffer| in |commandBuffers|:
                             1. Execute each command in |commandBuffer|.{{GPUCommandBuffer/[[command_list]]}}.
+                        1. For each |transfer| in |transfers|:
+                            1. Transfer the |transfer|.{{GPUResourceTransfer/buffers}} to
+                                |transfer|.{{GPUResourceTransfer/receivingQueue}}.
+                            1. Transfer the |transfer|.{{GPUResourceTransfer/textureSubresources}} to
+                                |transfer|.{{GPUResourceTransfer/receivingQueue}}.
                     </div>
             </div>
         </div>
-</dl>
-
-## <dfn interface>GPUFence</dfn> ## {#fence}
-
-<script type=idl>
-interface GPUFence {
-    GPUFenceValue getCompletedValue();
-    Promise<undefined> onCompletion(GPUFenceValue completionValue);
-};
-GPUFence includes GPUObjectBase;
-</script>
-
-### Creation ### {#fence-creation}
 
-<script type=idl>
-dictionary GPUFenceDescriptor : GPUObjectDescriptorBase {
-    GPUFenceValue initialValue = 0;
-};
-</script>
-
-<dl dfn-type=method dfn-for=GPUQueue>
-    : <dfn>createFence(descriptor)</dfn>
+    : <dfn>wait(queue, value)</dfn>
     ::
-        Creates a {{GPUFence}}.
+        Waits for the |queue| reaching |value| on the [=Queue timeline=]. If |value| is undefined,
```

Comment:
"Wait" sounds like it might cpu-block.
Maybe:
> Synchronizes subsequent |this|'s queue operations (e.g. submit, writeBuffer) to happen-after |queue|'s queue operations on, up to and including when `|queue.signal|` returned |value| (or smaller).

Reply:
Actually we should consider renaming `wait` to `synchronizeAfter` for clarity.

Reply:
I agree that the name `wait` may be a little misleading, even if it's specified to happen on the queue timeline.
In native API's `vkQueueWaitIdle` is more of "me wait for the queue" than "queue wait for something else". Therefore, it seems to me that `waitFor` would make this clear.

Reply:
+1 for `waitFor`